### PR TITLE
Add changes for goals, stretch goals, challenge level as well as stakeholders, environment for problem statement

### PR DESCRIPTION
**Pull request by calvynsiong**
## 🔗 Related Issue(s)

- https://github.com/TPGEngine/tpg/issues/27
- https://github.com/TPGEngine/tpg/issues/28

## 📝 Description

- Added stakeholder, environment and challenge level section to Problem Statement
- Included 4 goals and 2 stretch goals for Goals section in Problem Statement with reasoning and measurable outcomes to track

## ✅ How Has This Been Tested?

image

image

image

image

image

### CODE REVIEW

File: docs/ProblemStatementAndGoals/ProblemStatement.tex [hunk 1]
```diff
@@ -47,31 +47,107 @@ \subsection{Inputs and Outputs}
...
+    
+    Set up fully automated CI pipeline to ensure all code changes in the TPG framework are continuously tested, verified and follow proper guidelines
+
+
+\textbf{Measurable Outcomes: 
+} To accomplish this goal, there should be a complete integration of a CI tool such as Github Actions or Gitlab CI. The CI pipeline should be invoked when changes are made to the framework via new PRs and should not result in blocking errors (either accept or reject changes after CI pipeline analyzes changes). 
+
+\textbf{Reasoning: } Establishing a CI pipeline will streamline the development process, reduce human error, and improve code reliability. Automating testing ensures that the framework is stable with each update, promoting scalability and maintainability
+
+\item \textbf{ The TPG Framework should be well tested}
```
Comment (richardl2003) [inline docs/ProblemStatementAndGoals/ProblemStatement.tex:86]: Maybe I'd reword this to "standardize software development workflows". The scope of this is bigger as unit testing falls under it and we're giving them a framework to allow them to do their dev

File: docs/ProblemStatementAndGoals/ProblemStatement.tex [hunk 2]
```diff
@@ -47,31 +47,107 @@ \subsection{Inputs and Outputs}
...
+
+\textbf{Reasoning:} Introducing a well-structured testing suite is critical for ensuring the reliability and correctness of the TPG framework. Unit tests will verify that individual components behave as expected, and adding integration tests will confirm the system functions correctly as a whole. High test coverage ensures that potential issues are caught early, reducing future bugs and improving maintainability. This goal also aligns with the project’s emphasis on adopting best practices in software engineering.
+
+\item \textbf{The TPG framework must have an Agent-Environment Interface between itself and MuJoCo}
+
+Develop a robust interface that allows the TPG framework to communicate and control robotic agents within the MuJoCo physics engine.
+
+\textbf{Measurable Outcomes:
+} To accomplish this goal,  the TPG framework should contain a fully functional interface for a basic agent (e.g., a simple robotic arm) in MuJoCo. By the end of the project, this can be further expanded to having the interface to support more complex agents, such as quadrupeds.
+
```
Comment (richardl2003) [inline docs/ProblemStatementAndGoals/ProblemStatement.tex:102]: The robotic arm isn't necessarily the "basic agent". From my understanding the "programs" that control the robotic arms are the agent

File: docs/ProblemStatementAndGoals/ProblemStatement.tex [hunk 3]
```diff
@@ -47,31 +47,107 @@ \subsection{Inputs and Outputs}
...
+\textbf{Reasoning:} This interface will enable the TPG framework to be tested in more realistic and complex environments. Successful integration with MuJoCo will validate TPG’s performance in a cutting-edge robotics simulator, essential for showcasing its real-world capabilities.
+
+
+\item \textbf{The TPG Integration in MuJoCo should have determinable behavior when ran}
```
Comment (richardl2003) [inline docs/ProblemStatementAndGoals/ProblemStatement.tex:106]: I think I'd reword this goal

Some of the problems Dr. Kelly said we'd encounter are:
- Temporal Credit Assignment Problem (which is that sometimes things go bad first then become better)
- Agents need to have memory
- Current testing is in stationary environments, however MuJoCo enables testing in dynamic environments
- TPG is mean't to be an alternative to traditional Deep Learning NN's, maybe we can do an evaluation between TPG vs a more traditional DL NN in the MuJoCo environment to determine which is more performant? I know that the main draw to TPG is the cost, but can it outperform the alternative?
- Kelly wants to explore embedded evolution, could there be a goal which encompasses this?

File: docs/ProblemStatementAndGoals/ProblemStatement.tex [hunk 4]
```diff
@@ -47,31 +47,107 @@ \subsection{Inputs and Outputs}
...
+
+
+
+    \begin{enumerate}
+        \item \textbf{The TPG framework is integrated with additional Robotics Simulators}
+
+Extend the agent-environment interface to support additional simulators
+
+\textbf{Reasoning:} Supporting multiple simulators will increase the flexibility of TPG, allowing it to be tested in diverse environments. This would expand the framework’s use cases, enhancing its appeal to the wider RL and robotics communities.
+
```
Comment (richardl2003) [inline docs/ProblemStatementAndGoals/ProblemStatement.tex:132]: MuJoCo is pre industry standard from what I've seen. Maybe list out some more specific examples?

File: docs/ProblemStatementAndGoals/ProblemStatement.tex [hunk 5]
```diff
@@ -47,31 +47,107 @@ \subsection{Inputs and Outputs}
...
+To adapt the TPG’s algorithm to interact seamlessly with MuJoCo’s physics-based simulations. This requires not only designing a robust agent-environment interface but also ensuring that TPG can handle the increased computational demands and complexity of real-world-like environments. The lack of prior work connecting TPG with high-fidelity simulators adds to the difficulty, as there are no existing best practices or benchmarks for this integration.
+The project involves significant software engineering improvements, including the introduction of automated testing, continuous integration, and comprehensive documentation. These tasks require deep understanding of the existing C++ codebase, ensuring that the framework remains largely unaffected by the team’s changes while adhering to modern development standards.
+Given these factors, the project represents an ambitious endeavor, involving the understanding of cutting-edge research and development. 
+
```
Comment (richardl2003) [inline docs/ProblemStatementAndGoals/ProblemStatement.tex:150]: Dr. Smith said that our CL is "General", so we can't say "Advanced" just yet

**Review by richardl2003 — COMMENTED** [review]
@calvynsiong Left some comments

- Also the formatting for sections 1.3 vs 1.4 looks a bit off? Could u get those aligned?

File: docs/ProblemStatementAndGoals/ProblemStatement.tex (hunk 2 above)
Comment (calvynsiong) [reply docs/ProblemStatementAndGoals/ProblemStatement.tex:102]: Yea I guess I was thinking of what it will control in the end (Iikely a single joint model) but I'll make that distinction

**Review by richardl2003 — APPROVED** [review]
lgtm
